deb/openmediavault-bcache/usr/share/php/openmediavault/system/storage/bcache/backingdevice.inc

```diff
@@ -0,0 +1,255 @@
+	 * @throw \OMV\ExecException
+	 */
+	public function getState() {
+		if (!$this->isRegistered()) {
+			return 'unregistered';
+		}
+		if (!$this->isRunning()) {
+			return 'missing cache';
+		}
+		return $this->getValue("state");
+	}
+
+	/**
+	 * If bcache is running (i.e. whether the /dev/bcache device exists,
+	 * whether it’s in passthrough mode or caching.
+	 * @return boolean True if bcache is running, otherwise false.
+	 * @throw \OMV\ExecException
+	 */
+	public function isRunning() {
+		return $this->getValue("running") == 1;
+	}
+
+	/**
+	 * Shut down the bcache device and close the backing device.
+	 * @return void
+	 * @throw \OMV\ExecException
+	 */
+	public function stop() {
+		while ($this->isRegistered()) {
+			$this->setValue("stop", 1);
+			sleep(1);
+		}
+	}
+
+	/**
+	 * General function to get sys values.
+	 * @param string name of value.
+	 * @param boolean when true return only first line.
+	 * @return A output array.
+	 * @throw \OMV\ExecException
+	 */
+	public function getValue($name, $firstValue = TRUE) {
+		$filename = "/sys/block/{$this->getDeviceName(TRUE)}/bcache/{$name}";
+		$cmd = new \OMV\System\Process("cat", $filename);
+		$cmd->setQuiet(TRUE);
+		$cmd->execute($output);
+		if ($firstValue)
+			return $output[0];
+		return $output;
+	}
+
+	/**
+	 * General function to set values.
+	 * @param string name of value.
+	 * @param string|int the value.
+	 * @return void
+	 * @throw \OMV\ExecException
+	 */
+	public function setValue($name, $value) {
+		$filename = "/sys/block/{$this->getDeviceName(TRUE)}/bcache/{$name}";
+		$cmdLine = sprintf("echo %s > %s", $value, $filename);
+		$cmd = new \OMV\System\Process($cmdLine);
+		$cmd->setRedirect2to1();
+		$cmd->execute();
+	}
+
+	/**
+	 * Can be one of either writethrough, writeback, writearound or none.
+	 * @return string On of the cache modes.
+	 * @throw \OMV\ExecException
+	 */
+	public function getCacheMode() {
+		$output = $this->getValue("cache_mode");
+		$pattern = '{\[(\w+)\]}';
+		if (preg_match($pattern, $output, $matches)) {
+			return $matches[1];
+		}
+		return $output;
+	}
+
+	/**
+	 * A sequential IO will bypass the cache once it passes this threshold.
+	 * @return integer threshold in bytes.
+	 * @throw \OMV\ExecException
+	 */
+	public function getSequentialCutoff() {
+		$output = $this->getValue("sequential_cutoff");
+		$pattern = '{(\d*(?:\.\d+)?)([kMG])}';
+		$value = 0;
+		$unit = "";
+		if (preg_match($pattern, $output, $matches)) {
+			$value = (float) $matches[1];
+			$unit = $matches[2];
+		}
+		switch ($unit) {
+            case 'G':
+                $value *= 1024;
+            case 'M':
+                $value *= 1024;
+			case 'k':
+                $value *= 1024;
+            default:
+                break;
+        }
+		$value = ceil($value);
+		return $value;
+	}
+
+	/**
+	 * Get statistic of the bcache device.
+	 * @return The statistic of the bcache device.
+	 * @throw \OMV\ExecException
+	 */
+	public function getStatistic() {
```

Comment:
Is it possible to return JSON data here. This would make it easier to understand what each value is used for.

Reply:
Do you have an example how to map json to datatable?
The json would look like this:
`
{
  "bypassed" => {
     "fiveminutes" => 10,
     "hour" => 20,
     ...
  },
  "cache_bypass_hits" => {
     "fiveminutes" => 10,
     ...
  }
}
`

Reply:
> Do you have an example how to map json to datatable?

Do not understand the question.

Reply:
OK, after reviewing the backend and frontend code we can close the discussion. Everything is fine.

Reply:
I have made the following changes to return json data:
https://github.com/myrck/openmediavault/commit/68873f023a8baea0db953111e223432a0e53fb2f

And the other side is the frontend `omv-storage-bcache-backing-details-tabs-page`, what changes are necessary to display the json structure in the data table? Or should I convert back to an object in the rpc class like this
https://github.com/openmediavault/openmediavault/blob/89e5c98bc3e79d28c3f09249fcc6e09269825579/deb/openmediavault-podman/usr/share/openmediavault/engined/rpc/podman.inc#L178

Reply:
It was in another branch, i will delete this.